be1-go/channel/digitalCash/mod.go

```diff
@@ -0,0 +1,227 @@
+	}
+
+	err = c.handleMessage(publish.Params.Message, socket)
+	if err != nil {
+		return xerrors.Errorf("failed to handle a publish message: %v", err)
+	}
+
+	return nil
+}
+
+// Catchup is used to handle a catchup message.
+func (c *Channel) Catchup(catchup method.Catchup) []message.Message {
+	c.log.Info().Str(msgID, strconv.Itoa(catchup.ID)).Msg("received a catchup")
+
+	return c.inbox.GetSortedMessages()
+}
+
+// Broadcast is used to handle a broadcast message.
+func (c *Channel) Broadcast(broadcast method.Broadcast, socket socket.Socket) error {
+	c.log.Info().Msg("received a broadcast")
+
+	err := c.verifyMessage(broadcast.Params.Message)
+	if err != nil {
+		return xerrors.Errorf("failed to verify broadcast message: %w", err)
+	}
+
+	err = c.handleMessage(broadcast.Params.Message, socket)
+	if err != nil {
+		return xerrors.Errorf("failed to handle broadcast message: %v", err)
+	}
+
+	return nil
+}
+
+// ---
+// Message handling
+// ---
+
+// handleMessage handles a message received in a broadcast or publish method
+func (c *Channel) handleMessage(msg message.Message, socket socket.Socket) error {
+	err := c.registry.Process(msg, socket)
+	if err != nil {
+		return xerrors.Errorf("failed to process message: %w", err)
+	}
+
+	c.inbox.StoreMessage(msg)
+
+	err = c.broadcastToAllClients(msg)
+	if err != nil {
+		return xerrors.Errorf("failed to broadcast message: %v", err)
+	}
+
+	return nil
+}
+
+// NewDigitalCashRegistry creates a new registry for the digital cash channel
+func (c *Channel) NewDigitalCashRegistry() registry.MessageRegistry {
+	registry := registry.NewMessageRegistry()
+
+	registry.Register(messagedata.TransactionPost{}, c.processTransactionPost)
+
+	return registry
+}
+
+// broadcastToAllClients is a helper message to broadcast a message to all clients.
+func (c *Channel) broadcastToAllClients(msg message.Message) error {
+	c.log.Info().Str(msgID, msg.MessageID).Msg("broadcasting message to all")
+
+	rpcMessage := method.Broadcast{
+		Base: query.Base{
+			JSONRPCBase: jsonrpc.JSONRPCBase{
+				JSONRPC: "2.0",
+			},
+			Method: "broadcast",
+		},
+		Params: struct {
+			Channel string          `json:"channel"`
+			Message message.Message `json:"message"`
+		}{
+			c.channelID,
+			msg,
+		},
+	}
+
+	buf, err := json.Marshal(&rpcMessage)
+	if err != nil {
+		return xerrors.Errorf("failed to marshal broadcast query: %v", err)
+	}
+
+	c.sockets.SendToAll(buf)
+
+	return nil
+}
+
+// verifyMessage checks if a message in a Publish or Broadcast method is valid
+func (c *Channel) verifyMessage(msg message.Message) error {
+	jsonData, err := base64.URLEncoding.DecodeString(msg.Data)
+	if err != nil {
+		return xerrors.Errorf("failed to decode message data: %v", err)
+	}
+
+	// Verify the data
+	err = c.hub.GetSchemaValidator().VerifyJSON(jsonData, validation.Data)
+	if err != nil {
+		return xerrors.Errorf("failed to verify json schema: %w", err)
+	}
+
+	c.log.Info().Str(msgID, msg.MessageID).Msg("!!!NEWMSG!!!: " + msg.Data)
```

Comment:
I imagine this was for debugging purposes, otherwise the logging msg seems a bit weird :)